src/core/kconfigelektra.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Created by felix on 28.10.19.
+//
+
+#include "kconfigelektra.h"
+#include "kconfigdata.h"
+
+#include <utility>
+#include <iostream>
+
+using namespace kdb;
+
+KConfigElektra::KConfigElektra(std::string appName, uint majorVersion, std::string profile) : app_name(std::move(
+        appName)), major_version(majorVersion), profile(std::move(profile)) {}
+
+KConfigElektra::KConfigElektra(std::string appName, uint majorVersion) :
+        KConfigElektra::KConfigElektra(std::move(appName), majorVersion, "current") {}
+
+KConfigElektra::~KConfigElektra() = default;
+
+/**
+ * Move both name iterators to the point, where they differ.
+ *
+ * @param parent the iterator of the parent key
+ * @param child the iterator of the child key
+ */
+inline void traverseIterators(Key::iterator* parent, Key::iterator* child) {
+    /*
+     * First item of iterator seems to be namespace, as namespaces can differ with cascading keys, we skip these.
+     * (Loop initializer)
+     *
+     * Loop as long as the paths are the same and iterate through the path segments.
+     */
+    for ((*parent)++, (*child)++; parent->get() == child->get(); (*parent)++, (*child)++);
+}
+
+struct KConfigKey {
+    std::string group;
+    std::string key;
+};
+
+inline KConfigKey elektraKeyToKConfigKey(Key::iterator* key, Key::iterator* end) {
```

Comment:
If you're not intending to manipulate an existing iterator, it would be safer to pass as value. Key::iterator (or NameIterator) only contains 3 pointers, so it won't introduce any noticeable overhead